File: CodingSamples/Java/Language/InterfaceTest1/src/Banker.java
package banking;

public class Banker {

	private static long seed;

	//static code block is executed once when the class
	//is used for the first time by the executing program
	//allowing this class to initialize its static fields 
	static {
		seed = System.currentTimeMillis() % 100000000L;
	}

	public static Account openCurrentAccount() {
		var acc = new CurrentAccount(); //var = CurrentAccount
		acc.id = ++seed + 1000000000L;
		return acc;
	}

	public static Account openSavingsAccount() {
		var acc = new SavingsAccount();
		acc.id = ++seed + 2000000000L;
		return acc;
	}

	//private constructor to stop instantiation of this class
	//since all of its members are static
	private Banker() {}
}

File: CodingSamples/DotNet/Language/AbstractClassTest/BankLib/Banker.cs
namespace Banking;

//a class declared with static modifier only contains
//static members and such does not support a constructor
public static class Banker
{
    private static long seed;

    //a static constructor is invoked once when
    //the type is referenced for the first time
    //within the executing application
    static Banker()
    {
        seed = DateTime.Now.Ticks % 1000000;
    }

    public static Account OpenCurrentAccount()
    {
        var acc = new CurrentAccount(); //var = CurrentAccount
        acc.Id = ++seed + 100000000L;
        return acc;
    }

    public static Account OpenSavingsAccount()
    {
        SavingsAccount acc = new();
        acc.Id = ++seed + 200000000L;
        return acc;
    }

    //An extension method is a member of a static class 
    //whose first parameter is declared with 'this' keyword.
    //Such a method can be called as an instance method
    //of its first parameter type by using namespace of its class
    public static void LockAccount(this Account target)
    {
        target.Id = -target.Id;
    }
}